hg-fast-export.py

```diff
@@ -311,6 +316,9 @@ def get_branchname(name):
     removed=[r.decode(fn_encoding).encode('utf8') for r in removed]
 
   removed=[strip_leading_slash(x) for x in removed]
+  
+  if '.hgsubstate' in added+removed+changed:
```

Comment:
This check seems like it could lead to a significant slowdown. You are scanning through all changed files for each commit. As you know what the file name is, you should be able to look it up in the manifest (which uses dictionaries) and determine if it has changed in O(1) instead of O(N).

Reply:
Yeah, that was quite crude fix. I tried to make as little changes in code as possible due to my lack of knowledge in python and, I suppose, internals of mercurial.

I'll try to use manifest, but it'll require more thorough review :)